azurerm/internal/services/network/tests/resource_arm_express_route_gateway_test.go

```diff
@@ -0,0 +1,238 @@
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_express_route_gateway" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.ExpressRouteGatewaysClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMExpressRouteGateway_basic(data acceptance.TestData) string {
+	template := testAccAzureRMExpressRouteGateway_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = "acctestER-gateway-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+  scale_units         = 1
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMExpressRouteGateway_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMExpressRouteGateway_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = azurerm_express_route_gateway.test.name
+  resource_group_name = azurerm_express_route_gateway.test.name
+  location            = azurerm_express_route_gateway.test.location
+  scale_units         = 1
+}
+`, template)
+}
+
+func testAccAzureRMExpressRouteGateway_updateScaleUnits(data acceptance.TestData, scaleUnits int) string {
+	template := testAccAzureRMExpressRouteGateway_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = "acctestER-gateway-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+  scale_units         = %d
+}
+`, template, data.RandomInteger, scaleUnits)
+}
+
+func testAccAzureRMExpressRouteGateway_tags(data acceptance.TestData) string {
+	template := testAccAzureRMExpressRouteGateway_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_gateway" "test" {
+  name                = "acctestER-gateway-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+  scale_units         = 1
+  tags = {
+    Hello = "World"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMExpressRouteGateway_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctestvwan-%d"
```

Comment:
COuld we use some caps here
```suggestion
  name                = "acctest-VWAN-%d"
```